### ISSUE TITLE
Convert Async Await to ESTree format

### ISSUE THREAD

**Issue by flying-sheep** [eslint/typescript-eslint-parser#119]
* **TypeScript:** 2.1.1
* **`typescript-eslint-parser`:** 1.0.0

using eslint with this parser on that:

```typescript
await somthing()
```

i got

> [eslint] Expected an assignment or function call and instead saw an expression. (no-unused-expressions)

that bug was fixed in eslint some time ago: eslint/eslint#7175

is it possible that you don’t emit a `Await` for expressions like this? because if you do, this should be treated like a call/yield, and ignored by this rule

**Comment by flying-sheep** [comment]
hmm, seems like it’s a

```typescript
ExpressionStatement {
  expression: TSAwaitExpression { ... }
}

**Comment by JamesHenry** [comment]
Yes, we don't have specific conversion for async await yet.

I am going to wait on the final resolution of https://github.com/eslint/espree/pull/305 and then port the async await test suite from espree to this project

cc @nzakas
